File: src/primus/pawel/Exercise_200.java
package primus.pawel;
//String Task
// 7kyu
// https://www.codewars.com/kata/598ab63c7367483c890000f4/train/java
// 2021-08-25 T:12:25:28
class Exercise_200 {

    public static String perform(String word) {
        StringBuilder resultSB = new StringBuilder();
        resultSB.append(".");
        for (int i = 0; i < word.length(); i++) {
            if(!characterIsVowel(word.charAt(i))){
                resultSB.append(word.charAt(i)).append(".");
            }
        }

        return resultSB.deleteCharAt(resultSB.length()-1).toString().toLowerCase();
    }

    public static boolean characterIsVowel(Character character){
        Character lowerCaseCharacter = Character.toLowerCase(character);
        return lowerCaseCharacter == 'a' ||lowerCaseCharacter == 'o' || lowerCaseCharacter == 'y'
                || lowerCaseCharacter == 'e' || lowerCaseCharacter == 'u' ||lowerCaseCharacter == 'i';
    }
}

File: src/JS/Exercise_200.js
// Word to binary
// 7kyu
// https://www.codewars.com/kata/59859f435f5d18ede7000050/train/javascript
// 2022-08-13 T:14:41:20

function wordToBin(str) {
    return [...str].map(i => charToBinary(i));
}

function charToBinary(char) {

    var nr = char.charCodeAt(0);
    return '0' + nr.toString(2);
}

console.log(charToBinary('A'))
console.log(wordToBin('AB'))
